hspec-core/src/Test/Hspec/Core/Extension/Config/Type.hs

```diff
@@ -0,0 +1,184 @@
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
+module Test.Hspec.Core.Extension.Config.Type (
+  Config(..)
+, to
+, from
+
+, setAnnotation
+, getAnnotation
+
+, setSpecTransformation
+, getSpecTransformation
+
+, applySpecTransformation
+) where
+
+import           Prelude ()
+import           Test.Hspec.Core.Compat
+
+import           Test.Hspec.Core.Format
+import           Test.Hspec.Core.Config.Definition (ColorMode(..), UnicodeMode(..))
+import qualified Test.Hspec.Core.Config.Definition as Core
+import           Test.Hspec.Core.Annotations (Annotations)
+import qualified Test.Hspec.Core.Annotations as Annotations
+
+import           Test.Hspec.Core.Extension.Tree (SpecTree)
+
+data Config = Config {
```

Comment:
I think it's possible to remove some of those options after #870.  However, only from this `Config` type. The fields of the core `Config` can only be deprecated, not removed, for backwards compatibility.

That's also the reason why I duplicated the type, so that the extension API doesn't "know  about" legacy stuff.

It probably makes sense to finish #870 before the extensions API lands  on `main`.  

That's the approach I'm taking for now:
- The  provisional API is on #894, I use this as a staging ground
- #870 will go on #894 once it's done
- Any additional changes will be merged to #894
- Once everything looks good #894 will go on `main`


This PR is now against #894, with the options and documentation changes.

Reply:
Thank your for outlining the dev plan for extensions. Is there anything you want me to specifcally look at?